src/pykoala/corrections/sky.py

```diff
@@ -734,6 +868,211 @@ def combine_telluric_corrections(list_of_telcorr, ref_wavelength):
         
+    def qc_plots(self, show=False, save_as=None):
+        '''
+        Top: Filtered intensity for every fibre.
+        Centre: Sky-subtracted.
+        Bottom: Filtered sky.
+        '''
+        fig, axes = new_figure('wavelet_filter', nrows=3, ncols=2, sharey=False, gridspec_kw={'width_ratios': [1, .02], 'hspace': 0., 'wspace': .1})
+        im, cb = colour_map(fig, axes[0, 0], 'wavelet coefficient', self.filtered, x=self.wavelength, ylabel='spec_id', cbax=axes[0, 1])
+        im, cb = colour_map(fig, axes[1, 0], 'sky-subtracted', self.filtered - self.sky*self.sky_weight, x=self.wavelength, ylabel='spec_id', cbax=axes[1, 1])
+        ax = axes[-1, 0]
+        axes[-1, 1].axis('off')
+        ax.plot(self.wavelength, self.sky, 'k-', alpha=.5, label='sky (median coefficient)')
+        ax.fill_between(self.wavelength, self.sky_lo, self.sky_hi, color='k', alpha=0.1, label='uncertainty (16-84%)')
+        ax.plot(self.wavelength, self.sky/self.sky_weight, 'r:', alpha=.5, label='unweighted sky')
+        ax.set_xlabel('wavelength [pix]')
+        ax.set_ylabel(f'filtered (scale = {self.scale} pix)')
+        ax.legend()
+        if show:  # TODO: deal with ipympl
+            fig.show()
+        if save_as is not None:
+            fig.savefig(save_as)
+
+        
+    def get_throughput_object(self):
+        return Throughput(throughput_data=np.repeat(self.fibre_throughput[:, np.newaxis], self.wavelength.size + 3*self.scale, axis=1))
+
+    
+
+class SkySelfCalibration(CorrectionBase):
+    """Wavelength calibration, throughput, and sky model based on strong sky lines."""
+    name = "SkySelfCalibration"
+    verbose = True
+
+    # TODO: Don't assume RSS format (intensity[spec_id, wavelength])
+    #       def __init__(self, dc:DataContainer, continuum:ContinuumModel):
+    def __init__(self, dc:RSS):
+        self.update(dc)
+
+    def update(self, dc:RSS):
+        self.dc = dc
+        self.biweight_sky()
+        self.update_sky_lines()
+        self.calibrate()
+
+
+    # TODO: use SkyModel as an argument to update()?
+    def biweight_sky(self):
+        #self.wavelength = self.dc.wavelength.to_value(u.Angstrom)
+        self.sky_intensity = stats.biweight.biweight_location(self.dc.intensity, axis=0)
+
+
+    def update_sky_lines(self):
+        sky_cont, sky_err = ContinuumEstimator.lower_envelope(self.dc.wavelength, self.sky_intensity)#, min_separation)
+        sky_lines = self.sky_intensity - sky_cont
+
+        self.continuum = ContinuumModel(self.dc)
+        self.continuum.strong_sky_lines.add_column(0.*u.Angstrom, name='sky_wavelength')
+        self.continuum.strong_sky_lines.add_column(0., name='sky_intensity')
+        for line in self.continuum.strong_sky_lines:
+            wavelength = self.dc.wavelength[line['left']:line['right']]
+            spectrum = sky_lines[line['left']:line['right']]
+            weight = spectrum**2
+            line['sky_wavelength'] = np.nansum(weight*wavelength) / np.nansum(weight)
```

Comment:
This produces an error in cell 19 of the tutorial notebook due to different units (length and dimensionless).